### PR TITLE
[AIRFLOW-XXXX] Experimental Airflow native executor

### PR DESCRIPTION
**Pull request by nuclearpinguin**
This is experimental work on Airflow native executor based on kombu.

### CODE REVIEW

File: airflow/executors/worker_executor.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                # retry_policy={
+                #     'interval_start': 0,  # First retry immediately,
+                #     'interval_step': 2,  # then increase by 2s for every retry.
+                #     'interval_max': 3,  # but don't exceed 3s between retries.
+                #     'max_retries': 1,   # give up after 1 tries.
+                # },
+            )
+            self.queued_tasks.pop(key)
+            # TODO: do we need to keep this information in executor?
+            # self.running.add(key)
```
Comment (nuclearpinguin) [inline airflow/executors/worker_executor.py]: It seems that this is only used for printing progress. But I am not convinced that updating progress is something worth to implement  bidirectional communication.

**Comment by nuclearpinguin** [comment]
So, it works. Steps to play:
1. Open 3 terminals in breeze
2. In one run `airflow consumer start`, in second `watch -n "ps -aux"`
3. In third do:
- `export AIRFLOW__WORKER__BROKER_URL=redis://redis:6379/`
- ` export AIRFLOW__CORE__EXECUTOR=AirflowExecutor`
- `airflow dags backfill example_bash_operator -s 2019-12-06`

🙀

File: airflow/executors/worker_executor.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        pass
+
+    def execute_async(
+        self,
+        key: TaskInstanceKeyType,
+        command: CommandType,
+        queue: Optional[str] = None,
+        executor_config: Optional[Any] = None,
+    ):
+        """Do not allow async execution for Celery executor."""
```
Comment (potiuk) [inline airflow/executors/worker_executor.py]: Change it to native :)

Comment (nuclearpinguin) [reply airflow/executors/worker_executor.py]: Yeah, I have to fix many things and decided on the name... Airflow, Native, Kombu, Consumer Executor?

**Comment by potiuk** [comment]
Really nice. I also looked and there seem to be WIP on Google PubSub transport for Kombu : https://github.com/celery/kombu/issues/705 so if we implement it, we could use Kombu as a common interface for whatever Queuing system we want

**Comment by potiuk** [comment]
I think we should bring it forward to other commiters :).

**Comment by nuclearpinguin** [comment]
I will polish that (types, comments, docs) and we can open draft to Apache and discuss there.

**Pull request closed without merging**